core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1841,28 +1841,20 @@ RETURNING "txes".*
 		if err != nil {
 			return pkgerrors.Wrap(err, "CreateEthTransaction failed to insert evm tx")
 		}
-		var pruned int64
-		pruned, err = txRequest.Strategy.PruneQueue(ctx, o)
-		if err != nil {
-			return pkgerrors.Wrap(err, "CreateEthTransaction failed to prune evm.txes")
-		}
-		if pruned > 0 {
-			o.logger.Warnw(fmt.Sprintf("Dropped %d old transactions from transaction queue", pruned), "fromAddress", txRequest.FromAddress, "toAddress", txRequest.ToAddress, "meta", txRequest.Meta, "subject", txRequest.Strategy.Subject(), "replacementID", dbEtx.ID)
-		}
```

Comment:
I think I understand the motivation to remove this from the `CreateTransaction` tx_store method based on your conversation with Patrick but worried if there're are any implications. With the existing logic if pruning the queue failed, the transaction would fail to create. The new behavior would allow the transaction to be created even if prune fails. Sounds like that would be ok but don't know if I'm missing any context on why it was made dependent like this in the first place. Maybe it was to prevent the queue from consistently growing if pruning is failing?

Reply:
Yes, nice catch.
I think the current changes are good.
Pruning happens first, and then Tx creation. If pruning fails, we fail the TxCreation, thus preventing the queue to grow more.
If Tx creation fails, the pruning would still apply. Not a big deal.